ContestModel/src/org/icpc/tools/contest/model/internal/account/PublicContest.java

```diff
@@ -0,0 +1,185 @@
+package org.icpc.tools.contest.model.internal.account;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.icpc.tools.contest.model.IAccount;
+import org.icpc.tools.contest.model.IClarification;
+import org.icpc.tools.contest.model.IContestObject;
+import org.icpc.tools.contest.model.IGroup;
+import org.icpc.tools.contest.model.IJudgement;
+import org.icpc.tools.contest.model.IProblem;
+import org.icpc.tools.contest.model.IState;
+import org.icpc.tools.contest.model.ISubmission;
+import org.icpc.tools.contest.model.ITeam;
+import org.icpc.tools.contest.model.ITeamMember;
+import org.icpc.tools.contest.model.internal.Account;
+import org.icpc.tools.contest.model.internal.Contest;
+
+/**
+ * Public filter: information that everyone can see, regardless of role (including teams and
+ * spectators) or whether they've logged in.
+ *
+ * This filter removes:
+ * <ul>
+ * <li>Accounts (but subclasses can set username to allow their own account through)</li>
+ * <li>Problems before contest start</li>
+ * <li>Hidden groups & teams (and all associated data)</li>
+ * <li>Clarifications (except broadcasts)</li>
+ * <li>Judgemewnts and commentary after the freeze</li>
+ * <li>Test_data</li>
+ * <li>Runs</li>
+ * <li>Commentary</li>
+ * </ul>
+ */
+public class PublicContest extends Contest {
+	protected String username;
+
+	protected List<IProblem> problems = new ArrayList<IProblem>();
+
+	public PublicContest() {
+		// do nothing
+	}
+
+	@Override
+	public void add(IContestObject obj) {
+		IContestObject.ContestType cType = obj.getType();
+
+		switch (cType) {
+			// all of these are fully public
+			case CONTEST:
+			case JUDGEMENT_TYPE:
+			case MAP_INFO:
+			case START_STATUS:
+			case AWARD:
```

Comment:
Are awards not updated during the freeze? The spec says the freeze should be taken into account. Imagine a CCS updating awards during the freeze, that would not be OK if the public sees that, right?

Reply:
If you take a look at the code before, awards weren't even output to roles below blue/staff. Maybe I should just hide them again now and do the full logic via another PR? Or ok to defer this fix?